src/storage/mvcc/txn.rs

```diff
@@ -428,7 +429,7 @@ impl<S: Snapshot> MvccTxn<S> {
 
         // Rollback is called only if the transaction is known to fail. Under the circumstances,
         // the rollback record needn't be protected.
-        self.cleanup(key, TimeStamp::zero(), false)
+        cleanup(self, key, TimeStamp::zero(), false)
```

Comment:
I didn't expect that some action is called by `MvccTxn`. So could you explain what's the relationship between actions and `MvccTxn` now?

Reply:
> I didn't expect that some action is called by `MvccTxn`. So could you explain what's the relationship between actions and `MvccTxn` now?

In fact `rollback` itself should be an action too, but I'm not very sure why we still need a `rollback`  if it just forward it to a simple `cleanup` call, no matter it is a method in `MvccTxn` or is an action.

Reply:
Now I think it's not necessary that command and action are in a one-to-one relationship. I think we can have no "rollback" action. Rollback command calls the cleanup action.